plugins/modules/hdbuserstore.py

```diff
@@ -0,0 +1,273 @@
+  become: true
+  become_user: <hanasid>adm
+"""
+
+RETURN = r"""
+key:
+    description: HDB key name
+    type: str
+    returned: always
+    sample: mykey
+env:
+    description: HDB env name
+    type: str
+    returned: When state is C(present)
+    sample: myenv
+username:
+    description: HDB username for key
+    type: str
+    returned: When state is C(present)
+    sample: myusername
+"""
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def ensure_created(module):
+    binary = os.path.join(module.params.get("binary_path"), "hdbuserstore")
+    key = module.params.get("key")
+    env = module.params.get("env")
+    username = module.params.get("username")
+    password = module.params.get("password")
+    force = module.params.get("force")
+    return_key = {"key": key, "env": env, "username": username}
+
+    # Fetch the key:
+    rc, stdout, _err = module.run_command(args=[binary, "List", key])
+    if rc not in [0, 100]:
+        module.fail_json(msg="Failed to execute list: {0}".format(stdout))
+
+    if rc == 0 and not force:
+        return False, stdout, return_key
+
+    # Store the key:
+    if not module.check_mode:
+        rc, stdout, _err = module.run_command(
+            args=[binary, "Set", key, env, username, password]
+        )
+        if rc != 0:
+            module.fail_json(msg="Failed to execute Set: {0}".format(stdout))
+
+    return True, stdout, return_key
+
+
+def ensure_absent(module):
+    binary = os.path.join(module.params.get("binary_path"), "hdbuserstore")
+    key = module.params.get("key")
+
+    rc, stdout, _err = module.run_command(args=[binary, "List", key])
+    if not module.check_mode and rc == 0:
+        rc, _out, _err = module.run_command(args=[binary, "Delete", key])
+
+    if rc not in [0, 100]:
+        module.fail_json(msg="Failed to execute delete: {0}".format(stdout))
+
+    return rc == 0, stdout, {"key": key}
+
+
+def main():
+    module_args = dict(
+        state=dict(
+            type="str",
+            choices=[
+                "present",
+                "absent",
+            ],
+            default="present",
+        ),
+        key=dict(type="str", required=True, no_log=False),
+        env=dict(type="str"),
+        username=dict(type="str"),
+        password=dict(type="str", no_log=True),
+        binary_path=dict(type="str", default=""),
+        force=dict(type="bool", default=False),
+    )
+
+    result = dict(changed=False)
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["env", "username", "password"]],
+        ],
+    )
+
+    state = module.params.get("state")
+    if state == "present":
+        result["changed"], result["stdout"], return_key = ensure_created(module)
+    else:
+        result["changed"], result["stdout"], return_key = ensure_absent(module)
+
+    result.update(return_key)
```

Comment:
what's the value of this command ? in my opinion line 262 or 265 should update the return_key